### PR TITLE
[PG-1036] Add events support to the GPHVideoView in RN SDK - iOS

### PR DESCRIPTION
**Pull request by pshoniuk**

### CODE REVIEW

**Review by shwetaatgiphy — APPROVED** [review]
hi @pshoniuk 

I only read the changes to the doc & those make sense to me. 
The rest of the changes I'm good if Ivan/Alex are cool with them. 


Thank you for the tag

File: ios/RNGiphyVideoView.swift [hunk 1]
```diff
@@ -52,34 +59,65 @@ class RNGiphyVideoView: UIView {
   
   private func updateVolume() -> Void {
     DispatchQueue.main.async {
+      if self.videoView.media == nil {
```
Comment (ALexanderLonsky) [inline ios/RNGiphyVideoView.swift:62]: np:
`guard self.videoView.media != nil else return`
or
`guard let _ = self.videoView.media else return`

File: ios/RNGiphyVideoView.swift [hunk 2]
```diff
@@ -52,34 +59,65 @@ class RNGiphyVideoView: UIView {
...
       } else {
         self.videoView.unmute()
       }
     }
   }
-  
+
   private func updatePlaying() -> Void {
     DispatchQueue.main.async {
+      if self.videoView.media == nil {
```
Comment (ALexanderLonsky) [inline ios/RNGiphyVideoView.swift:76]: np:
guard would be better

File: ios/RNGiphyVideoView.swift [hunk 3]
```diff
@@ -52,34 +59,65 @@ class RNGiphyVideoView: UIView {
...
   }
-  
+
   //MARK: RN Properties
   @objc func setMedia(_ rnValue: NSDictionary) -> Void {
     GPHMedia.fromRNValue(rnValue) { self.media = $0 }
   }
   
   @objc func setPlaying(_ value: Bool) -> Void {
+    if self.playing == value {
```
Comment (ALexanderLonsky) [inline ios/RNGiphyVideoView.swift:94]: np:
guard would be better

File: ios/RNGiphyVideoView.swift [hunk 4]
```diff
@@ -52,34 +59,65 @@ class RNGiphyVideoView: UIView {
...
+      return
+    }
     self.playing = value
   }
   
   @objc func setMuted(_ value: Bool) -> Void {
+    if self.muted == value {
```
Comment (ALexanderLonsky) [inline ios/RNGiphyVideoView.swift:101]: np:
guard would be better

**Review by ALexanderLonsky — APPROVED** [review]

**Review by ALexanderLonsky — APPROVED** [review]

**Pull request merged**